### ISSUE TITLE
Calling removeDefinition not affect list of classes

### ISSUE THREAD

**Issue by romanmatyus** [nette/di#66]
```php
class MyExtension extends CompilerExtension
{
	function loadConfiguration()
	{
		$builder->addDefinition($this->prefix('myClass'))
			->setFactory('My\Class');
		$builder->removeDefinition($this->prefix('myClass'));
		$builder->findByType('My\Class');
	}
}
```
This returning unexpected notice ```E_NOTICE: Undefined index: my.myClass``` when calling ```$builder->findByType('My\Class')``` in [ContainerBuilder on line (220)](http://api.nette.org/2.3.0/source-DI.ContainerBuilder.php.html#220).

**Comment by pilec** [comment]
Can you make pool with failing test with this logic?

**Comment by dg** [comment]
Closed #71
